contracts/integrations/test/forwarder/forwarder_test.ts

```diff
@@ -0,0 +1,597 @@
+import { DummyERC20TokenContract } from '@0x/contracts-erc20';
+import { DummyERC721TokenContract } from '@0x/contracts-erc721';
+import {
+    artifacts as exchangeArtifacts,
+    BlockchainBalanceStore,
+    ExchangeContract,
+    LocalBalanceStore,
+} from '@0x/contracts-exchange';
+import { artifacts, ForwarderContract } from '@0x/contracts-exchange-forwarder';
+import {
+    blockchainTests,
+    constants,
+    expect,
+    getLatestBlockTimestampAsync,
+    getPercentageOfValue,
+    toBaseUnitAmount,
+} from '@0x/contracts-test-utils';
+import { assetDataUtils, ForwarderRevertErrors } from '@0x/order-utils';
+import { BigNumber } from '@0x/utils';
+
+import { Actor, actorAddressesByName, FeeRecipient, Maker } from '../actors';
+import { deployForwarderAsync } from './deploy_forwarder';
+import { ForwarderTestFactory } from './forwarder_test_factory';
+import { DeploymentManager } from '../utils/deployment_manager';
+
+blockchainTests('Forwarder integration tests', env => {
+    let deployment: DeploymentManager;
+    let forwarder: ForwarderContract;
+    let balanceStore: BlockchainBalanceStore;
+    let testFactory: ForwarderTestFactory;
+
+    let makerToken: DummyERC20TokenContract;
+    let makerFeeToken: DummyERC20TokenContract;
+    let anotherErc20Token: DummyERC20TokenContract;
+    let erc721Token: DummyERC721TokenContract;
+    let nftId: BigNumber;
+    let wethAssetData: string;
+    let makerAssetData: string;
+
+    let maker: Maker;
+    let taker: Actor;
+    let orderFeeRecipient: FeeRecipient;
+    let forwarderFeeRecipient: FeeRecipient;
+
+    before(async () => {
```

Comment:
Optional: The situation in this `before` is **vastly** better than it was before. This said, there is still a lot of code here---albeit much simpler code than before. I think that we could probably do more work in the constructor of `maker` and `taker` like configuring token balances for some tokens as an option. It's out of scope of this PR, but I'm curious to hear any thoughts that you might have on this.

If we do implement anything like this, we'll need to use a pattern like `deployAsync` from the `DeploymentManager` so that we can use `async/await` syntax.

Reply:
hmm,, interesting idea –– not sure if the `deployAsync` pattern would play nicely with the mixin pattern, but I'll look into it